lib/api/apiUtils/bucket/bucketCors.js

```diff
@@ -318,3 +318,32 @@ export function parseCorsXml(xml, log, cb) {
         return cb(null, validationRes);
     });
 }
+
+export function convertToXml(arrayRules) {
+    const xml = [];
+    xml.push('<?xml version="1.0" encoding="UTF-8" standalone="yes"?>',
+    '<CORSConfiguration>');
+    arrayRules.forEach(rule => {
+        xml.push('<CORSRule>');
+        ['allowedMethods', 'allowedOrigins', 'allowedHeaders', 'exposeHeaders']
+        .forEach(elementArr => {
+            if (rule[elementArr]) {
+                const element = elementArr.charAt(0).toUpperCase() +
+                    elementArr.slice(1, -1);
+                rule[elementArr].forEach(value => {
+                    xml.push(`<${element}>${escapeForXml(value)}` +
+                    `</${element}>`);
+                });
+            }
+        });
+        if (rule.id) {
+            xml.push(`<ID>${escapeForXml(rule.id)}</ID>`);
```

Comment:
AWS does not restrict the values for rule.id so it may contain symbols in addition to alphanumeric characters.
The value I don't escape is rule.maxAgeSeconds which is validated so it must be a number.